pkg/odo/cli/url/create.go

```diff
@@ -241,12 +286,18 @@ func NewCmdURLCreate(name, fullName string) *cobra.Command {
 	}
 	urlCreateCmd.Flags().IntVarP(&o.urlPort, "port", "", -1, "port number for the url of the component, required in case of components which expose more than one service port")
 	urlCreateCmd.Flags().BoolVarP(&o.secureURL, "secure", "", false, "creates a secure https url")
-	// if experimental mode is enabled, add more flags to support ingress creation based on devfile
+	// if experimental mode is enabled, add more flags to support ingress creation or docker application based on devfile
 	if experimental.IsExperimentalModeEnabled() {
-		urlCreateCmd.Flags().StringVar(&o.tlsSecret, "tls-secret", "", "tls secret name for the url of the component if the user bring his own tls secret")
-		urlCreateCmd.Flags().StringVarP(&o.host, "host", "", "", "Cluster ip for this URL")
+		if pushtarget.IsPushTargetDocker() {
+			urlCreateCmd.Flags().IntVarP(&o.exposedPort, "exposed-port", "", -1, "an external port to the application container")
```

Comment:
How is this flag going to work for K8s and OpenShift?
If this is valid only for local than the description should mention that.

Reply:
We could also consider hiding flags based on `pushtarget` settings.

Reply:
this flag is only added if `pushtarget.IsPushTargetDocker()`. It won't be valid for kube cases.

Reply:
I missed the `if` statement, or it was not there before?